File: 3185. Count Pairs That Form a Complete Day II.java
class Solution {
    public long countCompleteDayPairs(int[] hours) {
        Map<Integer, Integer> freq = new HashMap<>();
        long answer = 0;
        for (int i = 0; i < hours.length; i++) {
            int value = hours[i] % 24;
            answer += freq.getOrDefault((24-value)%24, 0);
            freq.put(value, freq.getOrDefault(value, 0) + 1);
        }
        return answer;
    }
}

File: 3185. Count Pairs That Form a Complete Day II.go
func countCompleteDayPairs(hours []int) int64 {
    freq := make(map[int]int)
    var count int64 = 0
    for _, i := range hours {
        i %= 24
        if val, ok := freq[(24-i)%24]; ok {
            count += int64(val)
        }
        freq[i] += 1
    }
    return count
}
